test/variable_env_test.py

```diff
@@ -1,36 +1,109 @@
-from pettingzoo.test.api_test import api_test
+import pytest
+
+from pettingzoo.test import api_test, parallel_api_test, seed_test
 from pettingzoo.test.example_envs import (
     generated_agents_env_v0,
     generated_agents_parallel_v0,
 )
-from pettingzoo.test.parallel_test import parallel_api_test
-from pettingzoo.test.seed_test import check_environment_deterministic
+from pettingzoo.test.seed_test import (
+    check_environment_deterministic,
+    check_environment_deterministic_parallel,
+    parallel_seed_test,
+)
 from pettingzoo.utils.conversions import aec_to_parallel, parallel_to_aec
 
 
 def test_generated_agents_aec():
     api_test(generated_agents_env_v0.env())
-    # seed_test(generated_agents_env_v0.env)
+    seed_test(generated_agents_env_v0.env)
 
 
 def test_generated_agents_parallel():
     parallel_api_test(generated_agents_parallel_v0.parallel_env())
+    parallel_seed_test(lambda: generated_agents_parallel_v0.parallel_env())
+
+
+@pytest.mark.skip(
+    reason="parallel_to_aec wrapper produces nondeterministic behavior with this environment."
+)
+def test_generated_agents_parallel_to_aec():
     api_test(generated_agents_parallel_v0.env())
     # seed_test(generated_agents_parallel_v0.env)
-    # seed_test(generated_agents_parallel_v0.env)
 
 
 def test_parallel_generated_agents_conversions():
     parallel_api_test(aec_to_parallel(generated_agents_parallel_v0.env()))
     api_test(parallel_to_aec(generated_agents_parallel_v0.parallel_env()))
 
+    # double-wrapping should produce the same behavior as unwrapped env
+    env1 = generated_agents_parallel_v0.parallel_env()
+    env2 = aec_to_parallel(parallel_to_aec(generated_agents_parallel_v0.parallel_env()))
+
+    # fails (due to parallel_to_aec conversion issues with this environment)
+    # check_environment_deterministic_parallel(env1, env2, 500)
+
+    # double-wrapping should produce deterministic behavior
+    parallel_seed_test(
+        lambda: aec_to_parallel(
+            parallel_to_aec(generated_agents_parallel_v0.parallel_env())
+        )
+    )
+
+    # double- and quadruple- wrapping should cancel out in the same way
+    env1 = aec_to_parallel(parallel_to_aec(generated_agents_parallel_v0.parallel_env()))
+    env2 = aec_to_parallel(
+        parallel_to_aec(
+            aec_to_parallel(
+                parallel_to_aec(generated_agents_parallel_v0.parallel_env())
+            )
+        )
+    )
+    assert type(env1) == type(
+        env2
+    ), "parallel_to_aec and aec_to_parallel wrappers should cancel out. Result: {type(env1)}, {type(env2)}."
```

Comment:
We don't need this. Instead just check that double wrapped is same as unwrapped.

Reply:
I'm leaving these in just in case something goes wrong in the future and we have to debug and figure out what's wrong with this environment, good to have these baselines to ensure that wrapping these different number of times works correctly (given that one of the tests that is still failing is double wrapped vs unwrapped, and triple vs single wrapped was a separate problem and has been fixed now)